phongo_compat.h

```diff
@@ -92,9 +92,17 @@
 #  define ARG_UNUSED
 #endif
 
-#if SIZEOF_LONG == 4
+#if PHP_VERSION_ID < 70000
+typedef long zend_long;
+# define SIZEOF_ZEND_LONG SIZEOF_LONG
+#endif
+
+#if SIZEOF_ZEND_LONG == 8
+# define ADD_INDEX_INT64(zval, index, value) add_index_long(zval, index, value)
+# define ADD_ASSOC_INT64(zval, key, value) add_assoc_long(zval, key, value)
```

Comment:
These definitions were moved up to be consistent with `bson.c`, where we check for 64-bit platforms first. Additionally, I removed the trailing semi-colon from `ADD_ASSOC_INT64()`'s definition for consistency.